pulsar-client/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java

```diff
@@ -89,12 +89,27 @@
     /**
      * Specify a list of topics that this consumer will subscribe on.
      * <p>
-     * This argument is required when constructing the consumer.
      *
      * @param topicNames
      */
     ConsumerBuilder topics(List<String> topicNames);
 
+    /**
+     * Specify a pattern for topics that this consumer will subscribe on.
+     * <p>
+     *
+     * @param topicsPattern
+     */
+    ConsumerBuilder topicsPattern(Pattern topicsPattern);
+
+    /**
+     * Specify a pattern for topics that this consumer will subscribe on.
```

Comment:
Can you explain a bit more in detail what kind of pattern is accepted (say regex/glob .. ) and also provide a short example snippet on how to use this.

Reply:
Thanks, will add it.